### PR TITLE
Refresh the Swift Package Registry OpenAPI document

### PR DESCRIPTION
**Pull request by czechboy0**
### Motivation:

The OpenAPI document for Swift Package Registry was incomplete and didn't contain all the requirements described in the related Swift Evolution proposals.

### Modifications:

Rewrote the OpenAPI document based on the current reading of all the Swift Package Registry Swift Evolution proposals and documentation in this repository.

You can view the rendered version of the docs at: https://editor.swagger.io/?url=https://raw.githubusercontent.com/czechboy0/swift-package-manager/refs/heads/hd-refresh-registry-openapi/Documentation/PackageRegistry/registry.openapi.yaml

### Result:

A much more accurate OpenAPI doc, one that can be used for code generation.

### CODE REVIEW

**Comment by dschaefer2** [comment]
@swift-ci please test

**Review by dschaefer2 — APPROVED** [review]
Awesome. Thanks @czechboy0!

File: Documentation/PackageRegistry/registry.openapi.yaml [hunk 1]
```diff
@@ -1,448 +1,703 @@
...
+      type: object
+      description: A problem detail object conforming to RFC7807
+      properties:
+        type:
+          type: string
+          format: uri
+          description: |-
+            A URI reference that identifies the problem type.
+            This specification encourages that, when dereferenced, 
+            it provide human-readable documentation for the problem type.
```
Comment (plemarquand) [inline Documentation/PackageRegistry/registry.openapi.yaml:587]:
```suggestion
            it provides human-readable documentation for the problem type.
```

**Review by plemarquand — APPROVED** [review]

**Comment by dschaefer2** [comment]
So weird why
@swift-ci please test windows
doesn't happen by default all the time...

**Comment by dschaefer2** [comment]
@swift-ci please test windows

**Review by jakepetroules — APPROVED** [review]

**Pull request merged**